__test__/helper/db-handler.js

```diff
@@ -36,4 +39,14 @@ module.exports.clearDatabase = async () => {
         const collection = collections[key];
         await collection.deleteMany();
     }
+}
+
+module.exports.refreshDatabase = async () => {
```

Comment:
- Please modify this method name to `refreshOpenDataCourses(){...}` or something similar to clarify that this only refreshes the course database (for reusability).
- This method uses OpenDataCourseRepository.batchCreateCourse(){...}` which bypasses dropping the table. This would create duplicates in the opendatacourses collection.
- Would it not be possible to just use `OpenDataCourseRepository.refreshCourseData(){...}` instead of the creation of this extra method? That would be better for maintainability (any change in the OpenDataCourseRepository methods responsible for the refresh would reflect in the tests without having to modify this method to reflect the changes).

Reply:
> Would it not be possible to just use OpenDataCourseRepository.refreshCourseData(){...} instead of the creation of this extra method? That would be better for maintainability (any change in the OpenDataCourseRepository methods responsible for the refresh would reflect in the tests without having to modify this method to reflect the changes).

No, for some reason Jest would not wait for the data course creation to finish before starting the tests until I used request() specifically.

Reply:
> This method uses OpenDataCourseRepository.batchCreateCourse(){...}` which bypasses dropping the table. This would create duplicates in the opendatacourses collection.

This is done once at the start and then the test table is dropped, so this can't happen. Remember that this is in memory virtual mongo server

Reply:
> > Would it not be possible to just use OpenDataCourseRepository.refreshCourseData(){...} instead of the creation of this extra method? That would be better for maintainability (any change in the OpenDataCourseRepository methods responsible for the refresh would reflect in the tests without having to modify this method to reflect the changes).
> 
> No, for some reason Jest would not wait for the data course creation to finish before starting the tests until I used request() specifically.

That's odd. Thanks for letting me know.